src/platform/net/zephyr/http.c

```diff
@@ -231,7 +231,7 @@ mender_http_perform(char                *jwt,
 
     mender_free(request.recv_buf);
 
-    return ret;
+    return (MENDER_FAIL != ret) ? ret : MENDER_RETRY_ERROR;
```

Comment:
If this part fails:
```

    /* Perform HTTP request */
    if ((http_req_ret = http_client_req(sock, &request, MENDER_HTTP_REQUEST_TIMEOUT, (void *)&request_context)) < 0) {
        mender_log_error("HTTP request failed: %s", strerror(-http_req_ret));
        goto END;
    }
```
`ret` will be `MENDER_FAIL` because the above call doesn't modify it in any way. And that's exactly where a network error would surface/happen, I believe.

Reply:
Bad reading of the code, sorry for the noise.